test/Microsoft.Azure.ServiceBus.UnitTests/ExpectedMessagingExceptionTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Xunit;
+
+    public class ExpectedMessagingExceptionTests
+    {
+        [Fact]
+        async Task MessageLockLostExceptionTest()
+        {
+            const int messageCount = 2;
+            var queueClient = QueueClient.CreateFromConnectionString(
+                TestUtility.GetEntityConnectionString(Constants.NonPartitionedQueueName));
+
+            await TestUtility.SendMessagesAsync(queueClient.InnerSender, messageCount);
+            var receivedMessages = await TestUtility.ReceiveMessagesAsync(queueClient.InnerReceiver, messageCount);
+
+            Assert.True(receivedMessages.Count() == messageCount);
+
+            // Let the messages expire
+            await Task.Delay(TimeSpan.FromMinutes(1));
```

Comment:
That should work too. Since this was a pretty important scenario, I wrote the actual use case. If time of test run becomes too long, I'll remember to change this.